onboard/nav/stateMachine.cpp

```diff
@@ -480,37 +489,86 @@ double StateMachine::getOptimalAvoidanceDistance() const
     return mPhoebe->roverStatus().obstacle().distance + mRoverConfig[ "navThresholds" ][ "waypointDistance" ].GetDouble();
 } // optimalAvoidanceAngle()
 
+// Returns if the waypoint is reachable without any avoidance.
 bool StateMachine::isWaypointReachable( double distance )
 {
     return isLocationReachable( mPhoebe, mRoverConfig, distance, mRoverConfig["navThresholds"]["waypointDistance"].GetDouble());
-} // isWaypointReachable
+} // isWaypo
 
+// Returns whether or not to enter RadioRepeaterTurn state.
 // If we have not already begun to drop radio repeater
-//  (RadioRepeaterTurn or RadioRepeaterDrive)
-// We should drop the repeater
-// and we haven't already dropped one
+// (RadioRepeaterTurn or RadioRepeaterDrive), it is time to drop the
+// repeater and we haven't already dropped one, then it's time to drop.
 bool StateMachine::isAddRepeaterDropPoint() const
 {
-
     return ( mPhoebe->roverStatus().currentState() != NavState::RadioRepeaterTurn &&
              mPhoebe->roverStatus().currentState() != NavState::RadioRepeaterDrive &&
-             mPhoebe->isTimeToDropRepeater() &&
-             mRepeaterDropComplete == false );
+             mIsTimeToDropRepeater &&
+             !mRepeaterDropComplete );
 } //isAddRepeaterDropPoint
 
-// Returns whether or not to enter RadioRepeaterTurn state.
+// Add the repeater drop point to the front of the path.
 void StateMachine::addRepeaterDropPoint()
 {
-    // TODO: signal drops before first completed waypoint
-    // Get last waypoint in path (where signal was good) and set search and gate
-    // to false in order to not repeat search
-    Waypoint way = (mPhoebe->roverStatus().course().waypoints)[mCompletedWaypoints-1];
-    way.search = false;
-    way.gate = false;
-
-    mPhoebe->roverStatus().path().push_front(way);
+    mRepeaterDropWaypoint.search = false;
+    mRepeaterDropWaypoint.gate = false;
+    mPhoebe->roverStatus().path().push_front(mRepeaterDropWaypoint);
 } // addRepeaterDropPoint
 
+// Executes the logic start counting number of iterations for how long it's been
+// since the rover has gotten a strong radio signal. If the signal drops
+// below the signalStrengthCutOff and the counting hasn't started, begin counting
+// iterations. Otherwise, the signal is good so the counting should be stopped.
+void StateMachine::updateRepeaterDropWaypoint()
+{
+    static unsigned int iterations = 0;
+
+    // If we haven't already dropped a repeater
+    // and our signal is above the threshold, start the iteration count
+    if( !mIsTimeToDropRepeater &&
```

Comment:
I don't follow why you care about this variable here based on your comment.

Reply:
My logic may be wrong but I don't want to update the point if we are currently driving to drop. Although, it would make the algorithm faster if we update the point along the way because then it would just drop it as soon as the signal is good.